packages/ui/src/components/AdvertisementTable.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  AdvertisementCA,
+  AdvertisementSubject,
+  Page,
+} from "@originator-profile/model";
+import { twMerge } from "tailwind-merge";
+import { _ } from "../utils";
+import Table from "./Table";
+import TableRow from "./TableRow";
+import { ExternalLink } from "./link";
+
+type Props = {
+  className?: string;
+  advertisement: AdvertisementCA;
+};
+
+function renderLinkRow(
+  ca: AdvertisementSubject,
+  key:
+    | "adReportContact"
+    | "adReviewGuidelines"
+    | "targetingPolicy"
+    | "adDataHandlingPolicy",
+  header: string,
+) {
+  const value = ca[key];
+  if (!value) return null;
+  return (
+    <TableRow
+      header={header}
+      data={<ExternalLink href={value.id}>{value.name}</ExternalLink>}
+    />
+  );
+}
+
+function renderAdDisplayRationale(rationale?: {
+  page?: Page;
+  description?: string;
+}) {
+  if (!rationale) return null;
+  if (rationale.page?.id && rationale.page?.name) {
+    return (
+      <>
+        <ExternalLink href={rationale.page?.id}>
+          {rationale.page?.name}
+        </ExternalLink>
```

Comment:
スキーマ定義で生成される型にしたがって実装すればよいかとおもいます。

この場合 Page 型は `{ id: string; name: string }` ですので `rational.page` の型から `undefined` でないことをチェックすればオプショナルチェーン (`?.`) は不要なはずです


```suggestion
  if (!rationale) return null;
  if (rationale.page) {
    return (
      <>
        <ExternalLink href={rationale.page.id}>
          {rationale.page.name}
        </ExternalLink>
```

Reply:
承知しました。
スキーマ定義で生成される型にしたがっての実装に修正します！